lib/whitehall_importer/integrity_checker/unpublishing_check.rb

```diff
@@ -0,0 +1,56 @@
+module WhitehallImporter
+  class IntegrityChecker::UnpublishingCheck
+    attr_reader :edition, :unpublishing
+
+    def initialize(edition, unpublishing)
+      @edition = edition
+      @unpublishing = unpublishing
+    end
+
+    def expected_type?
+      unpublishing["type"] == expected_type
+    end
+
+    def expected_type
+      return "withdrawal" if edition.withdrawn?
+      return "invalid edition state" unless edition.removed?
+
+      edition.status.details.redirect? ? "redirect" : "gone"
+    end
+
+    def expected_unpublishing_time?
+      return true unless edition.withdrawn?
+
+      unpublishing_time_matches?
+    end
+
+    def expected_unpublishing_time
+      edition.status.details&.withdrawn_at
+    end
+
+    def expected_alternative_path?
+      return true unless edition.removed?
+
+      unpublishing["alternative_path"] == edition.status.details.alternative_url
+    end
+
+    def expected_explanation?
+      Sanitize.clean(unpublishing["explanation"]).squish ==
+        Sanitize.clean(edition_explanation_html).squish
+    end
+
+  private
+
+    def unpublishing_time_matches?
+      unpublished_at = unpublishing["unpublished_at"]
+      Time.zone.iso8601(unpublished_at) == edition.status.details.withdrawn_at
```

Comment:
I've deployed the change to Publishing API now so this should be fine for rfc3339. I think we should aim to use the method @1pretz1 has, not sure how we sequence this though as you're both super close to finishing these PRs.

Reply:
He got there before me! 😄

Reply:
Went on a wild rebase journey with this one!  Will implement the time matching method next.